### DISCUSSION TITLE
5 Axis with STM32F401 Black Pill

### DISCUSSION THREAD

**Discussion by ArminMousavi** [grblHAL/core#463] [Q&A] [answered]
Hey, in the grblHAL Web Builder I can only select 4 axis for my machine on a STM32F401 controller but I want to add another axis and make the CNC a 5 axis, How can I do that ? what should I do ?

**Comment by terjeio** [comment]
You can use the CNC3040 board map? If not a new board map has to be added.

**Reply by ArminMousavi** [reply]
I'm not using any additional board, I wanna know if it's possible to do in normal stm32f401 black pill board ?

**Reply by terjeio** [reply, marked as answer]
The CNC3040 board map supports up to 5 axes. Click the _Board map_ button to see how pins are allocated

**Reply by ArminMousavi** [reply]
Thanks
